tests/gold_tests/pluginTest/header_rewrite/rules/rule_regex_sub.conf

```diff
@@ -0,0 +1,31 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Header:Test-Server-Sub extracts server version and type
+cond %{READ_RESPONSE_HDR_HOOK}
+cond %{HEADER:Server} /([^\/]*)(\/)(.*)/
+    set-header Test-Server-Sub $3:$1
+
+# HEADER:No-Regex has no regex pattern to match to
+cond %{SEND_RESPONSE_HDR_HOOK}
+cond %{NOW:YEAR} > 0
+    set-header Test-No-Regex No-Match-$1
+
+# HEADER:Test-Host-Sub replaces Host request header
+cond %{READ_RESPONSE_HDR_HOOK}
+cond %{HEADER:Host} /(\.)(.*)(\.)/
+    set-header Test-Host-Sub $1foo.$2$3
```

Comment:
I see in the code there are two places where it can now support $n, but I think the tests here only tests one of those cases? Under what condition does one use (and therefore test) the usage of $n inside of a "%{" ?

Reply:
@zwoop Should the regex substitutions be available in the header name as well? For example, `set-header $1 ...` which could then allow the usage of $n inside of a "%{}" like `set-header New-Header %{HEADER:$1}`